pdc_client/__init__.py

```diff
@@ -161,7 +258,13 @@ def decode(req):
 
         content_type = "application/json"
         encode = json.dumps
-        self.client = beanbag.BeanBag(url, session=self.session, fmt=(content_type, encode, decode))
+        # In get_paged function it returns an interator, so user can get each pages quickly in plugins.
+        # The original call function of BeanBag returns the list.
+        # Keep the current codes and options in codes. Just init MultiPageBeanBag when is_multi is true.
+        if is_multi:
+            self.client = MultiPageBeanBag(url, session=self.session, fmt=(content_type, encode, decode))
```

Comment:
Because if we use the new class here, the worker() in get_paged function should be removed as the new class can get all the pages.  

And as I comment on codes, the call function return a list, when you use pdc command to list data, the list function in plugins will get an interator return from get_pages.

As you mentioned, you prefer to let user quickly get first page when they list multi page. The worker() can do this thing but it can't add into the new class as there is a different return value.

So keep the original get_paged codes to let it return quickly when use pdc to get data by column but not json. 

And also I think we just provide a base class for users to use in there codes, and for our functions, we don't need to make change, because as you know for pdc and pdc_client  they
can get all pages.